File: devops/rodar/scripts/remove_redis.py
import os
import yaml

def remover_config_redis(caminho_arquivo):
    try:
        with open(caminho_arquivo, 'r', encoding='utf-8') as file:
            dados = yaml.safe_load(file)
        
        if dados is None:
            return

        dados.pop('redis', None)
        dados.pop('redisConfig', None)

        with open(caminho_arquivo, 'w', encoding='utf-8') as file:
            yaml.dump(dados, file, allow_unicode=True, default_flow_style=False, sort_keys=False)

        print(f"Configurações redis removidas de: {caminho_arquivo}")
    except yaml.YAMLError as e:
        print(f"Erro ao analisar YAML no arquivo: {caminho_arquivo} - {e}")
    except Exception as e:
        print(f"Erro ao processar {caminho_arquivo}: {e}")

def percorrer_diretorio(diretorio):
    print(f"Iniciando a verificação no diretório: {diretorio}")
    for root, _, files in os.walk(diretorio):
        for file in files:
            if file.endswith('.yaml') or file.endswith('.yml'):
                caminho_arquivo = os.path.join(root, file)
                print(f"Processando arquivo: {caminho_arquivo}")
                remover_config_redis(caminho_arquivo)
    print("Verificação concluída.")

if __name__ == "__main__":
    diretorio_alvo = '../applications' 
    if os.path.exists(diretorio_alvo) and os.path.isdir(diretorio_alvo):
        percorrer_diretorio(diretorio_alvo)
    else:
        print(f"Diretório especificado não existe ou não é um diretório: {diretorio_alvo}")

File: devops/rodar/scripts/remove_redis.js
const fs = require('fs');
const path = require('path');

const diretorioRaiz = '../applications'; 

function removerConfigRedis(caminho) {
    fs.readFile(caminho, 'utf8', (err, data) => {
        if (err) {
            console.error(`Erro ao ler arquivo: ${err}`);
            return;
        }

        const regexRedis = /redisConfig|redis/g;
        const novoConteudo = data.replace(regexRedis, '');

        fs.writeFile(caminho, novoConteudo, 'utf8', (err) => {
            if (err) {
                console.error(`Erro ao escrever arquivo: ${err}`);
                return;
            }
            console.log(`Configurações redis removidas de: ${caminho}`);
        });
    });
}

function percorrerDiretorio(diretorio) {
    fs.readdir(diretorio, (err, arquivos) => {
        if (err) {
            console.error(`Erro ao ler diretório: ${err}`);
            return;
        }

        arquivos.forEach((arquivo) => {
            const caminhoCompleto = path.join(diretorio, arquivo);
            fs.stat(caminhoCompleto, (err, stat) => {
                if (err) {
                    console.error(`Erro ao verificar arquivo: ${err}`);
                    return;
                }
                if (stat.isFile()) {
                    removerConfigRedis(caminhoCompleto);
                } else if (stat.isDirectory()) {
                    percorrerDiretorio(caminhoCompleto);
                }
            });
        });
    });
}

percorrerDiretorio(diretorioRaiz);
